cmd/starter/main_linux.go

```diff
@@ -21,44 +21,60 @@ import (
 	"github.com/sylabs/singularity/internal/app/starter"
 	"github.com/sylabs/singularity/internal/pkg/runtime/engines"
 	starterConfig "github.com/sylabs/singularity/internal/pkg/runtime/engines/config/starter"
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/imgbuild"
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/oci"
+	"github.com/sylabs/singularity/internal/pkg/runtime/engines/singularity"
 	"github.com/sylabs/singularity/internal/pkg/sylog"
 	_ "github.com/sylabs/singularity/internal/pkg/util/goversion"
 	"github.com/sylabs/singularity/internal/pkg/util/mainthread"
 )
 
-func getEngine(jsonConfig []byte) *engines.Engine {
-	engine, err := engines.NewEngine(jsonConfig)
-	if err != nil {
-		sylog.Fatalf("failed to initialize runtime: %s\n", err)
-	}
-	return engine
-}
-
 func startup() {
-	loglevel := os.Getenv("SINGULARITY_MESSAGELEVEL")
-	os.Clearenv()
-	if loglevel != "" {
-		if os.Setenv("SINGULARITY_MESSAGELEVEL", loglevel) != nil {
-			sylog.Warningf("can't restore SINGULARITY_MESSAGELEVEL environment variable")
-		}
-	}
-
+	// global variable defined in cmd/starter/c/starter.c
+	// C.config points to a shared memory area
 	cconf := unsafe.Pointer(C.config)
+	// initialize starter configuration
 	sconfig := starterConfig.NewConfig(starterConfig.CConfig(cconf))
+	// get JSON configuration originally passed from CLI
 	jsonConfig := sconfig.GetJSONConfig()
 
+	// get runtime engine name
+	name := engines.GetName(jsonConfig)
+	if name == "" {
+		sylog.Fatalf("no runtime engine selected")
+	}
+
+	// initialize corresponding runtime engine
+	if err := singularity.Init(name); err != nil {
+		sylog.Fatalf("%s", err)
```

Comment:
Should this be more verbose? I mean like: 

```suggestion
		sylog.Fatalf("Unable to start [...]: %s", err)
```